### PR TITLE
Embedding doc

### PR DESCRIPTION
**Pull request by jetfuel**
Reveal the high dimensional tab
Add function documentation
Improve loading animation algorithm
Use the in-house PCA function
Use the free T-SNE function

Resolve https://github.com/PaddlePaddle/VisualDL/issues/407

### CODE REVIEW

**Review by nickyfantasy — DISMISSED** [review]

File: visualdl/python/storage.py [hunk 1]
```diff
@@ -292,9 +295,19 @@ def text(self, tag):
         return self.writer.new_text(tag)
 
     def embedding(self):
+        """
+        Create an embedding writer that used to write
```
Comment (daming-lu) [inline visualdl/python/storage.py]: is used to

File: visualdl/python/storage.py [hunk 2]
```diff
@@ -292,9 +295,19 @@ def text(self, tag):
...
+        embedding data.
+
+        :return: A embedding writer to record embedding data
```
Comment (daming-lu) [inline visualdl/python/storage.py]: An embedding

File: visualdl/server/lib.py [hunk 3]
```diff
@@ -393,3 +392,23 @@ def _handler(key, func, *args, **kwargs):
         return data
 
     return _handler
+
+
+# A simple PCA implementaiton to do the dimension reduction.
```
Comment (daming-lu) [inline visualdl/server/lib.py]: Normally we comment methods like this:

```
def kos_root():
    """Return the pathname of the KOS root directory."""
    global _kos_root
    if _kos_root: return _kos_root
    ...
```

File: visualdl/server/lib.py [hunk 4]
```diff
@@ -393,3 +392,23 @@ def _handler(key, func, *args, **kwargs):
...
+def simple_pca(x, dimension):
+    # Center the data.
+    x -= np.mean(x, axis=0)
+
+    # Computing the Covariance Matrix
+    cov = np.cov(x, rowvar=False)
+
+    # Get eigenvectors and eigenvalues from the covariance matrix
+    eigvals, eigvecs = np.linalg.eig(cov)
```
Comment (daming-lu) [inline visualdl/server/lib.py:409]: What a math guy, Jeff! But do we need to do it ourselves? Moreover, SVD is more stable than eigenvector in terms of computing principle components. 

Check [here](https://stats.stackexchange.com/questions/314046/why-does-andrew-ng-prefer-to-use-svd-and-not-eig-of-covariance-matrix-to-do-pca)

File: visualdl/server/tsne.py [hunk 5]
```diff
@@ -0,0 +1,184 @@
+#
```
Comment (daming-lu) [inline visualdl/server/tsne.py:1]: We should give the original author a citation 😀 

`https://github.com/bhauman/neurpy/blob/master/tsne.py`

Comment (jetfuel) [reply visualdl/server/tsne.py:1]: I did give the author credit, I left my comments under the author's comments.

File: visualdl/server/lib.py (hunk 4 above)
Comment (jetfuel) [reply visualdl/server/lib.py:409]: The reason behind it so we don't have to import another pip package to do the calculation. 
And actually the TSNE file already has a clean PCA implementation. and I could just use that.

**Review by daming-lu — DISMISSED** [review]

**Review by daming-lu — APPROVED** [review]

**Pull request merged**